src/components/search-bar/search-bar.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useCallback, useState } from 'react';
+import { ReactComponent as CancelIcon } from '../../assets/svg/cancel-icon.svg';
+import { ReactComponent as SearchIcon } from '../../assets/svg/search-icon.svg';
+import './search-bar.scss';
+
+export interface SearchBarProps {
+  initialText?: string;
+  placeholder?: string;
+  disabled?: boolean;
+  onChange?: (value: string) => void;
+  onEnterPressed?: (value: string) => void;
+}
+
+export const SearchBar = ({
+  initialText,
+  placeholder,
+  disabled,
+  onChange,
+  onEnterPressed,
+}: SearchBarProps) => {
+  const [value, setValue] = useState(initialText);
+
+  const hasText = useCallback(() => {
```

Comment:
You're right on `hasText` the function gets recalculated when the useState changes anyway, so it's never actually memorized.

Memoizing event handlers is a good practice though since the only thing changing the function is realistically when the event handler itself is changed, so not re-initiating the function when the text changes is a good idea.

Reply:
Oh, I see. Ok, I'll start memoizing my event handlers then!

Reply:
On second thought, I think you're right that `useCallback` has more overhead than just creating the event handler function again. So, I'm just gonna remove all usages of it, it makes our code simpler anyway.